### ISSUE TITLE
APIError support for Status Code

### ISSUE THREAD

**Issue by birkagal** [achillean/shodan-python#213]
Hey there. 
When using this client, I found this weird behavoiur that the raised APIError contain only a spesific string for every type of error (502, 401 etc) and for others (like 404, which is pretty common..) it just add the text from the http response.. Which again is a very spesific string.

A much better approach is to also add the status code itself to the exception, and whenever the user catch the exception he can check for example if the status is 404 instead of checking that spesific string (which is what I'm currently doing 😃 ) .

So in short, I think we should add support for HTTP Status codes inside the APIError, whenever its appropiate.
There are cases we use APIError not when there is an actual HTTPError, so status_code can be None.

**Comment by Rajpatel143** [comment]
I'm really sry.
My frnd using is.
I dnt understand that.
I'm really sorry for trouble.
Plz I'm sry.
